compiler/luci/pass/src/RemoveUnnecessaryAddPass.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryAddPass.h"
+
+#include "helpers/NodeFiller.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+bool remove_no_effect_add(luci::CircleNode *node)
+{
+  auto target_node = dynamic_cast<luci::CircleAdd *>(node);
+  if (target_node == nullptr || target_node->dtype() != loco::DataType::FLOAT32)
+    return false;
+
+  //  NOTE for general activation function A: Act(A + 0) != A
+  if (target_node->fusedActivationFunction() != luci::FusedActFunc::NONE)
+    return false;
+
+  luci::CircleConst *const_operand = nullptr;
+  luci::CircleNode *nonconst_operand = nullptr;
+  if (not luci::fill(&const_operand, &nonconst_operand).with_commutative_args_of(target_node))
+    return false;
+
+  if (dynamic_cast<luci::CircleConst *>(nonconst_operand) != nullptr)
+  {
+    // NOTE this is degenerated case - (const1 + const2)
```

Comment:
```suggestion
    // NOTE this is degenerated '(const1 + const2)' case
```
?

Reply:
I mean that constant addition should be folded elsewhere.